### ISSUE TITLE
Extend react-native-desktop to support status-react Desktop functionality

### ISSUE THREAD

**Issue by MaxRis** [status-im/ideas#34]
## Preamble

    Idea: DEV#34
    Title: Extend react-native-desktop to support status-react Desktop functionality
    Status: Draft
    Created: 2017-11-22

## Summary
Status React cross-platform (Linux, OS X and Windows) desktop version based on [react-native-desktop](https://github.com/status-im/react-native-desktop/) port of react-native.

## Vision

React-Native-Desktop is the desktop extension of origin react-native framework developed by Status. With direct access to underlying OS's APIs (via Qt Framework APIs mainly) we should be able to have almost full control on software stack and provide better users experience.

Status IM application should has range of unique screens optimized for the best desktop users experience.

Desktops version of Status IM is supposed to re-use backend implementation of Status IM Mobile, but with unique UI screens and based on react-native-desktop extension of original react-native.

## Swarm Participants

- Lead Contributor: @MaxRis
- Testing & Evaluation: 
- Contributor: @vkjr
- Contributor: @flexsurfer
- UX (if relevant): 

## Goals & Implementation Plan

Goal Date: 2017-12-11 (Done)
Description: Enhance development cycle to get acceptable performance in code development (adopt figwheel + live-reload, find the way to compile ClojureScript to JS without code optimization (lein has known issues with that)

Goal Date: 2017-12-15 (Done)
Description: Login screen rendering and functionality

Goal Date: TODO
Description: Integrate react-native-desktop into latest status-react sources with desktops screens available.

Goal Date: TODO
Description: Resolve react-native-desktop related UI issues (missed support of some specific props or components, etc)

Goal Date: TODO
Description: Smoothly working figwheel for dev builds

Goal Date: TODO
Description: Stable Status Desktop application with screens specific for desktop users experience.

Goal Date: TODO
Description: Support by react-native-desktop of [all react-native components](https://github.com/status-im/status-electron/wiki/Status-Components are supported by react-native-desktop) used in Status Desktop.

Goal Date: TODO
Description: Automated process of Status Desktop distributions creations for Linux, OS X and Windows desktop platforms.

### Minimum Viable Product (completed)

1. Application shows 'chat' screen
2. User can login
3. User can exchange 1-1 messages

### Minimum Viable Product 2

Status Desktop application with screens specific for desktop users experience based on Status-React (Status IM Mobile) source code base with possibility to chat without interruption.

## Exit criterias

- All react-native components from the list of used components in Status Desktop https://github.com/status-im/status-electron/wiki/Status-Components are supported by react-native-desktop
- Existing UI for status-react desktop runs smoothly with `react-native-desktop`
- Automated creation of binary distributions for Linux, OS X, Windows desktop platforms.

## Success Metrics:

Successful internal usage by Status Team on daily basis.

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

**Comment by MaxRis** [comment]
@vkjr @narek-h 
To start discussion of Goals and Implementation plan, we might set following as first goals:
- Enhance development cycle to get acceptable performance (adopt figwheel + live-reload, find the way to compile ClojureScript to JS without code optimization (lein has known issues with that), build in parallel for mobile to be sure in compatibility ( ?  but it seems we need this only we can't have 100% code compatibility with available status-im code for mobile ))
- Make few ClojureScript code screens fully supported by Qt based port ( UI code + 3rd party plugins code ). Separate list of issues and not yet supported things by priority should be created.

**Comment by MaxRis** [comment]
Once we will have at least few screens of status-im fully supported by react-native-desktop, it might be good sign that react-native-desktop is ready for Desktop specific implementation with ClojureScript and status-im core team can try to start with Desktop version based on react-native-desktop.

**Comment by vkjr** [comment]
@MaxRis, agree with proposed steps

**Comment by vkjr** [comment]
@MaxRis, @narek-h, if we look at the code of different screens, we can find that every screen uses  quite a lot of 3rd party libraries and react native components.
For example `chat` screen (which would be nice to have early due to importance) depends on:
3rd party libraries: `svg, i18n, invertible-scroll-view, config, emojis, orientation, drawer, realm`
react-native components/APIs: `Alert, Animated view, dimensions `
So, it can take time until we see full chat screen. And if we measure progress in screens, we are going to be demoralised :)

Most of the 3rd-party libraries and react native components used by code in `components` and `utils` folders in `status-react`. `chat` screen depends on components: `vector-icons, status-bar, chat-icon, listview, toolbar-content, animation, list-selection`

So I would suggest to start supporting separate components and utils and track them as progress steps. They are more predictable. To support `vector-icon` component we need to implement `raeact-native-svg` library. To support `config` util we need to implement `react-native-config` library. And so on.

**Comment by vkjr** [comment]
Probably first components I would start with are: `vector-icons, i18n, chat-icon, action-button`

**Comment by flexsurfer** [comment]
hey @vkjr , what is `chat-icon`? and we don't need `action-button` for desktop

**Comment by MaxRis** [comment]
@vkjr @flexsurfer Idea is updated according to the latest discussions, please, feel free to contribute if something missed, etc, thanks!

**Comment by vkjr** [comment]
@MaxRis. thanks! I also updated "Title" in Preamble section to match the name of idea and added one more exit criteria - "Existing UI for status-react desktop runs smoothly with `react-native-desktop`
"

**Comment by oskarth** [comment]
Replaced by https://github.com/status-im/ideas/blob/master/ideas/34-react-native-qt.md
